recipes/dataframe/all/conanfile.py

```diff
@@ -152,7 +152,7 @@ def package_info(self):
             if Version(self.version) < "1.20.0" and not self.options.shared:
                 # weird but required in those versions of dataframe
                 self.cpp_info.defines.append("LIBRARY_EXPORTS")
-        if Version(self.version) >= "1.20.0" and self.options.shared:
+        if "1.20.0" <= Version(self.version) < "2.0.0" and self.options.shared:
```

Comment:
I see, it's a bug due to a refactoring where some stuff has been changed to templates. It should be patched instead of removing HMDF_SHARED in recipe.

Reply:
https://github.com/hosseinmoein/DataFrame/pull/235